src/redux/actions/index.ts

```diff
@@ -22,10 +32,20 @@ export function login(username : string) : LoginAction {
     }
 }
 
-export function addWorkout(payload : AddWorkoutPayload) {
+export function addWorkout(workoutName : string) : AddWorkoutAction  {
     return {
         type : actions.ADD_WORKOUT,
-        payload
+        payload : {
+            workoutName : workoutName
```

Comment:
I would suggest using workout id instead of name since multiple workouts can share the same name